### PR TITLE
Filter `_index` in advanced exports

### LINKED ISSUE #3333: Allow for `_index` to be filtered from exports
## Description

The custom exports feature allows for columns to be filtered but since the `_index` field is handled internally by formpack, it requires additional care.

### PR DESCRIPTION
**Pull request by joshuaberetta**
## Related issues

closes #3333

### CODE REVIEW

File: kpi/tests/test_mock_data_exports.py [hunk 1]
```diff
@@ -454,6 +454,16 @@ def test_xls_export_filter_fields(self):
         ]
         self.run_xls_export_test(expected_lines, export_options)
 
+    def test_xls_export_filter_fields_without_index(self):
+        export_options = {'fields': ["start", "end", "Do_you_descend_from_unicellular_organism"]}
+        expected_lines = [
+            [ "start", "end", "Do you descend from an ancestral unicellular organism?" ],
+            [ "2017-10-23T05:40:39.000-04:00", "2017-10-23T05:41:13.000-04:00", "No" ],
+            [ "2017-10-23T05:41:14.000-04:00", "2017-10-23T05:41:32.000-04:00", "No" ],
+            [ "2017-10-23T05:41:32.000-04:00", "2017-10-23T05:42:05.000-04:00", "Yes" ],
```
Comment (noliveleger) [inline kpi/tests/test_mock_data_exports.py:751]: Not a blocker at all, Just a notice that we could remove the leading and trailing spaces in each list ( and use single quotes ;-)). (Several other places in this file too)

**Review by noliveleger — COMMENTED** [review]
Because of a conflict with beta, I merged beta in your branch.
But the tests are not passing. 
> ======== 17 failed, 681 passed ...

Would you mind to look it ?

**Comment by noliveleger** [comment]
@joshuaberetta, as per our discussion, 17 tests are failing because it misses some mixes brought to `2.021.30`.
After merging `2.021.30`, only one test fails. This diff makes it pass though:

```diff 
--- a/kpi/tests/test_mock_data_exports.py
+++ b/kpi/tests/test_mock_data_exports.py
@@ -609,9 +609,10 @@ class MockDataExports(MockDataExportsBase):
     def test_xls_export_filter_fields_repeat_groups(self):
         export_options = {
             'fields': [
-                "_uuid",
-                "_submission_time",
-                "person/name",
+                '_uuid',
+                '_submission_time',
+                'person/name',
+                '_index',
             ]
         }
```
         
I would prefer if you could commit these changes (and test them) before I merge your PR.

**Review by noliveleger — CHANGES_REQUESTED** [review]
@joshuaberetta, as per our discussion, 17 tests are failing because it misses some fixes brought to `2.021.30`.
After merging `2.021.30`, only one test fails. This diff makes it pass though:

```diff 
--- a/kpi/tests/test_mock_data_exports.py
+++ b/kpi/tests/test_mock_data_exports.py
@@ -609,9 +609,10 @@ class MockDataExports(MockDataExportsBase):
     def test_xls_export_filter_fields_repeat_groups(self):
         export_options = {
             'fields': [
-                "_uuid",
-                "_submission_time",
-                "person/name",
+                '_uuid',
+                '_submission_time',
+                'person/name',
+                '_index',
             ]
         }
```
         
I would prefer if you could commit these changes (and test them) before I merge your PR, if you don't mind.

**Comment by joshuaberetta** [comment]
That was exactly the unpushed fix on my side too 🙃

**Pull request merged**